integreat_compass/cms/models/offers/offer_version.py

```diff
@@ -0,0 +1,114 @@
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy as _
+
+from ...constants import offer_version_states
+from ..abstract_base_model import AbstractBaseModel
+from .language import Language
+
+
+class OfferVersion(AbstractBaseModel):
+    """
+    Data model representing a Language.
+    """
+
+    offer = models.ForeignKey("Offer", on_delete=models.CASCADE)
```

Comment:
```suggestion
    offer = models.ForeignKey("Offer", on_delete=models.CASCADE, related_name="versions")
```
You could set a related name here to be able to address `offer.versions` directly instead of `offer.offerversions`.

Reply:
Hold on, isn't this already given by:

```
class Meta:
        verbose_name = _("offer version")
        verbose_name_plural = _("offer versions")
        default_related_name = "offer_versions"
        ordering = ["offer_version_date"]
        default_permissions = ("change", "delete", "view")
```

Reply:
No, the related name is the name of the attribute under which the reverse relationship is available...

https://docs.djangoproject.com/en/4.2/topics/db/queries/#backwards-related-objects
https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.ForeignKey.related_name

Reply:
Ah, you mean the `default_related_name`? Yes, this covers the same relationship, but `offer.versions` is still more concise than `offer.offer_versions`...
But yeah, doesn't make a huuuge difference :sweat_smile:

Reply:
> Ah, you mean the `default_related_name`? Yes, this covers the same relationship, but `offer.versions` is still more concise than `offer.offer_versions`... But yeah, doesn't make a huuuge difference sweat_smile

Yes :D

We could just rename the `default_related_version` to `versions`... It would probably still be clear enough...? Currently it's only relevant when accessing versions from the offer anyways.

Reply:
Hmm, the problem with that is that the default related name automatically applies to all foreign keys, so also `language` in this case...
And `language.versions` could be misleading, since it implies that it's a version of the language itself...
So I would suggest that the default related name is foreign-key-agnostic in a sense that it contains the whole model name, e.g. `language.offer_versions`, and only use the short related name `offer.versions` for the offer foreign key to avoid the duplication of the term "offer"...

Reply:
Alright, makes sense.